lib/ocpp/v201/charge_point.cpp

```diff
@@ -3817,6 +3820,36 @@ void ChargePoint::execute_change_availability_request(ChangeAvailabilityRequest
     }
 }
 
+// K01.27 - load profiles from database
+void ChargePoint::load_charging_profiles() {
+    try {
+        auto evses = this->database_handler->get_all_charging_profiles_by_evse();
+        EVLOG_info << "Found " << evses.size() << " evse in the database";
+        for (auto& profiles : evses) {
+            try {
+                auto evse_id = profiles.first;
+                for (auto profile : profiles.second) {
+                    if (this->smart_charging_handler->validate_profile(profile, evse_id) ==
+                        ProfileValidationResultEnum::Valid) {
+                        this->smart_charging_handler->add_profile(profile, evse_id);
+                    } else {
+                        // delete if not valid anymore
+                        this->database_handler->delete_charging_profile(profile.id);
+                    }
+                }
+            } catch (common::RequiredEntryNotFoundException& e) {
+                EVLOG_warning << "Could not get connector id from database: " << e.what();
+            } catch (const QueryExecutionException& e) {
+                EVLOG_warning << "Could not get connector id from database: " << e.what();
+            }
+        }
+    } catch (const QueryExecutionException& e) {
+        EVLOG_warning << "Could not load charging profiles from database: " << e.what();
+    } catch (const std::exception& e) {
+        EVLOG_warning << "Unknown error while loading charging profiles from database: " << e.what();
+    }
+}
```

Comment:
If the SmartChargingHandler has access to the database, and we're calling functions on the handler and modifying its state, couldn't this live in SmartChargingHandler?

Reply:
At the moment I was keeping this inline with what the community has in 1.6. Technically, we could defer to the `SmartChargingHandler`. Perhaps we could discuss a refactor at a later time?

Reply:
As we discussed we will keep it this way to keep the visibility of the method protected.